modules/ssh/ssh.go

```diff
@@ -55,10 +63,11 @@ func ScpFileToE(t *testing.T, host Host, mode os.FileMode, remotePath, contents
 	}
 	dir, file := filepath.Split(remotePath)
 
+	host.setDefaultPort()
```

Comment:
Forgetting to call this seems like a VERY likely gotcha we'll hit in the future. Any ideas on how to avoid that issue?

Reply:
👍 you kind of want it set upfront on `ssh.Host` instantiation. the only sane option I had was make sure it's called before the Port is used in the same method.

SO tells me the best options for constructors are: https://stackoverflow.com/a/18125763

- make the zero value a sensible default
- use a `New()` method

I'd be open to doing the `New() approach? I don't know if we can enforce using it though, without making `Host` private (a la `host`), but that means you can't touch it from your own codebase which is not optimal...